### PR TITLE
Rules for user and availabilities

### LINKED ISSUE #98: Define role access in database

### PR DESCRIPTION
**Pull request by Prolas**
Collection User:
Each user can read his own entry (see his personal information).
Admin can read/write.

Collection Availability:
Each user can read/write his own availability.
Admin can can read/write.

As the collections are not set up yet these rules are currently ignored as everybody can access everything.

close #98

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
-    // leaves your app open to attackers. At that time, all client
-    // requests to your Firestore database will be denied.
-    //
-    // Make sure to write security rules for your app before that time, or else
-    // all client requests to your Firestore database will be denied until you Update
-    // your rules
-    match /{document=**} {
-      allow read, write: if request.time < timestamp.date(2024, 4, 24);
+    	// Allow read, write if document exists and the user is owner or admin
+    	allow read, write: if isSignedIn() && resourceExists(resource) &&
```
Comment (mathieumognol) [inline firestore.rules:50]: What's the difference with lines 35 & 37?

File: firestore.rules [hunk 2]
```diff
@@ -1,19 +1,54 @@
...
+    function isAdmin(){
+    	return isOneOfRoles(["admin"])
+    }
+
+    match /user/{userId}{
+      // Allow read if user is owner
+    	allow read: if isSignedIn() && resourceExists(resource) && request.auth.uid == userId;
+      // Allow write if user is admin
+      allow write: if isSignedIn() && resourceExists(resource) &&
+      											(request.auth.uid == userId || isAdmin());
```
Comment (mathieumognol) [inline firestore.rules:38]: Why put both "(request.auth.uid == userId || isAdmin())" and not just "isAdmin()"?

**Review by mathieumognol — COMMENTED** [review]
Overall great rules to protect the database and give the good rights

File: firestore.rules (hunk 2 above)
Comment (Prolas) [reply firestore.rules]: This is indeed an error. I've also updated the rule so that the administrator has read access.

**Review by mathieumognol — APPROVED** [review]

**Pull request merged**